coresdk/src/backend/text_driver.cpp

```diff
@@ -36,61 +37,99 @@ void sk_finalize_text()
     TTF_Quit();
 }
 
-sk_font_data sk_load_font(const char * filename, int font_size)
+sk_font_data* sk_load_font(const char * filename, int font_size)
 {
     internal_sk_init();
 
-    sk_font_data font;
-    font.kind = SKFT_TTF;
-    font._data = TTF_OpenFont(filename, font_size);
-    if (!font._data)
+    sk_font_data *font = new sk_font_data;
+    font->id = FONT_PTR;
+    font->filename = filename;
+
+    sk_add_font_size(font, font_size);
+
+    return font;
+}
+
+void sk_add_font_size(sk_font_data *font, int font_size) {
+    if (VALID_PTR(font, FONT_PTR))
     {
-        cerr << "Error loading font " << SDL_GetError() << endl;
+        font->_data[font_size] = TTF_OpenFont(font->filename.c_str(), font_size);
+        if (!font->_data[font_size]) {
+            cerr << "Error loading font " << SDL_GetError() << endl;
+        }
     }
-    return font;
+    else
+    {
+        cerr << "Trying to load font size for an invalid font" << endl;
+    }
+}
+
+bool sk_contains_valid_font(sk_font_data* font)
+{
+    for (auto const it : font->_data) {
+        if (it.second) {
+            return true;
+        }
+    }
+
+    return false;
 }
 
 void sk_close_font(sk_font_data* font)
 {
-    if (font && font->_data)
+    if (VALID_PTR(font, FONT_PTR))
     {
-        TTF_CloseFont(static_cast<TTF_Font *>(font->_data));
-        font->kind = SKFT_UNKNOWN;
-        font->_data = NULL;
+        for (auto const it : font->_data) {
+            if (it.second) {
+                TTF_CloseFont(static_cast<TTF_Font *>(it.second));
+
+            }
+        }
+
+        font->name = "";
+        font->id = NONE_PTR;
+    }
+    else
+    {
+        cerr << "Trying to close font that is not a valid font." << endl;
     }
 }
 
-int sk_text_line_skip(sk_font_data* font)
+int sk_text_line_skip(sk_font_data* font, int font_size)
 {
-    if (font && font->_data)
-        return TTF_FontLineSkip(static_cast<TTF_Font *>(font->_data));
-    else
+    if (VALID_PTR(font, FONT_PTR)) {
+        return TTF_FontLineSkip(static_cast<TTF_Font *>(font->_data[font_size]));
```

Comment:
Need to check that this size is loaded as well... maybe add in a helper function that returns the TTF_Font* from a font + size. Then it can load etc... all in the one place.

It could return nullptr if it fails, so that would be all you need to check in each place.

The helper could check that the font was valid as well etc..